src/MerchantCenter/AccountService.php

```diff
@@ -623,4 +623,13 @@ public function update_wpcom_api_authorization( string $status, string $nonce ):
 	public function delete_wpcom_api_auth_nonce(): bool {
 		return $this->options->delete( OptionsInterface::GOOGLE_WPCOM_AUTH_NONCE );
 	}
+
+	/**
+	 * Check if the WPCOM API is approved.
+	 *
+	 * @return bool
+	 */
+	public function is_wpcom_api_approved(): bool {
```

Comment:
We do have already this method in [Options::is_wpcom_api_authorized](https://github.com/woocommerce/google-listings-and-ads/blob/develop/src/Options/Options.php#L215)

I don't mind one of another but it might need a little refactor. Maybe moving the check inside $this->oauth_service->revoke_wpcom_api_auth())